pkg/ingester/client/cortex.proto

```diff
@@ -20,11 +21,25 @@ service Ingester {
   rpc AllUserStats(UserStatsRequest) returns (UsersStatsResponse) {};
   rpc MetricsForLabelMatchers(MetricsForLabelMatchersRequest) returns (MetricsForLabelMatchersResponse) {};
 
-  // TransferChunks allows leaving ingester (client) to stream chunks directly to joining ingesters (server).
+  // TransferChunks allows leaving ingester (client) to stream chunks directly
+  // to joining ingesters (server). Called during the hand-off process.
   rpc TransferChunks(stream TimeSeriesChunk) returns (TransferChunksResponse) {};
 
   // TransferTSDB transfers all files of a tsdb to a joining ingester
   rpc TransferTSDB(stream TimeSeriesFile) returns (TransferTSDBResponse) {};
+
+  // SendChunks is invoked by a leaving ingester (client), streaming a subset
+  // of its chunks directly to an existing ingester.
+  rpc SendChunks(stream TimeSeriesChunk) returns (SendChunksResponse) {};
+
+  // GetChunks is invoked by a joining ingester (client). A subset of the
+  // serving ingester's chunks will be sent as a stream.
+  rpc GetChunks(GetChunksRequest) returns (stream TimeSeriesChunk) {};
+
+  // CleanupTransfer is invoked by a joining ingester after it is
+  // safe to perform cleanup on anything initialized by a call to
+  // GetChunks.
+  rpc CleanupTransfer(CleanupTransferRequest) returns (CleanupTransferResponse) {};
```

Comment:
This ends up just unblocking blocked ranges, but I'm hesitant to add any more context at the RPC level. I made it generic on purpose since Loki may do more than that. WDYT?